contracts/Intermediate_TokenRetriever.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.19;
+
+import "./Lockable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@uma/core/contracts/common/implementation/MultiCaller.sol";
+
+interface IBridgeAdapter {
+    function withdrawToken(
+        address recipient,
+        uint256 amountToReturn,
+        address l2Token
+    ) external;
+}
+
+/**
+ * @notice Contract deployed on an arbitrary L2 to act as an intermediate contract for withdrawals from L3 to L1.
+ * @dev Since each network has its own bridging requirements, this contract delegates that logic to a bridge adapter contract
+ * which performs the necessary withdraw action.
+ */
+contract Intermediate_TokenRetriever is Lockable, MultiCaller {
```

Comment:
I don't think we want `MultiCaller` here since we would rather have logic for many withdrawals on the bridge adapter contract and not on the retriever contract, since having it here means `n` delegatecalls for `n` internal transactions vs 1 delegatecall for `n` internal transactions (although removing it complicates the logic a bit).

Reply:
https://github.com/across-protocol/contracts/pull/605/commits/3131eed7f6f65691c99260b5529a43e54c499c21